pyclaragenomics/claragenomics/bindings/cudapoa.pyx

```diff
@@ -226,6 +229,43 @@ cdef class CudaPoaBatch:
         decoded_consensus = [c.decode('utf-8') for c in consensus]
         return (decoded_consensus, coverage, status)
 
+    def get_graphs(self):
+        """
+        Get the POA graph for each POA group.
+
+        Returns:
+            A tuple where
+            - first element is a networkx graph for each POA group
+            - second element is status of MSA generation for each group
+        """
+        cdef vector[DirectedGraph] graphs
+        cdef vector[cudapoa.StatusType] status
+        cdef vector[pair[DirectedGraph.edge_t, DirectedGraph.edge_weight_t]] edges
+        cdef DirectedGraph* graph
+        cdef DirectedGraph.edge_t edge
+        cdef DirectedGraph.edge_weight_t weight
+
+        # Get the graphs from batch object.
+        deref(self.batch).get_graphs(graphs, status)
+
+        nx_digraphs = []
+        for g in range(graphs.size()):
+            graph = &graphs[g]
+            edges = deref(graph).get_edges()
+            nx_digraph = nx.DiGraph()
+            for e in range(edges.size()):
+                edge = edges[e].first
+                weight = edges[e].second
+                nx_digraph.add_edge(edge.first,
+                                    edge.second,
+                                    weight=weight)
+            attributes = {}
+            for n in nx_digraph.nodes:
+                attributes[n] = {'label' : deref(graph).get_node_label(n).decode('utf-8')}
+            nx.set_node_attributes(nx_digraph, attributes)
+            nx_digraphs.append(nx_digraph)
+        return (nx_digraphs, status)
```

Comment:
This is just an idea: If we expose our Graph class to python anyways, why don't we return just that and provide a function to convert it to an nx.DiGraph() separately? At larger scales it might save the user an expensive construction, if the additional features of networkx are not needed.

Reply:
@ahehn-nv For python bindings there are two levels of exposing - one is where the C++ class is exposed in the binding, and the other is a custom python class that wraps that binding to expose to normal python usage. Right now I only implemented the first, which let's other bindings use the data type. The reason I didn't want to add another python class wrapping the graph class was mostly a maintainability thing. The primary use case for looking at the graph is for debugging, so the copy overhead won't be much of an issue. And users have more complex debugging functions available out of the box without having to copy things themselves. In time I'd like to maybe even replace our C++ graph API to a cuGraph compatible thing.

P.S. - I chose networkx as the interface coz it seemed widely used, had the dot serialization function, and I saw that used as a common graph lib at the hackathon :).

Reply:
Ok, makes sense. Thanks for the explanation. :)